src/wp-includes/media.php

```diff
@@ -1914,45 +1922,97 @@ function wp_filter_content_tags( $content, $context = null ) {
 }
 
 /**
- * Adds `loading` attribute to an `img` HTML tag.
+ * Adds optimization attributes to an `img` HTML tag.
  *
- * @since 5.5.0
+ * @since 6.3.0
  *
  * @param string $image   The HTML `img` tag where the attribute should be added.
  * @param string $context Additional context to pass to the filters.
- * @return string Converted `img` tag with `loading` attribute added.
+ * @return string Converted `img` tag with optimization attributes added.
  */
-function wp_img_tag_add_loading_attr( $image, $context ) {
-	// Get loading attribute value to use. This must occur before the conditional check below so that even images that
-	// are ineligible for being lazy-loaded are considered.
-	$value = wp_get_loading_attr_default( $context );
-
-	// Images should have source and dimension attributes for the `loading` attribute to be added.
+function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	// Images should have source and dimension attributes for the loading attributes to be added.
 	if ( false === strpos( $image, ' src="' ) || false === strpos( $image, ' width="' ) || false === strpos( $image, ' height="' ) ) {
```

Comment:
```suggestion
	if ( ! str_contains( $image, ' src="' ) || ! str_contains( $image, ' width="' ) || ! str_contains( $image, ' height="' ) ) {
```

Reply:
This snippetted is repeated a lot. A util function for this?

Reply:
It's only in two places, so not so bad. I would advise not to implement a utility function, mostly because it would be global due to WP's lack of using classes, so we can't really implement anything private. For something trivial like this, not worth having another global function.

But +1 on using `str_contains()`, will update.